### PR TITLE
openlineage, ftp: add OpenLineage support for FTPFileTransferOperator

### PR DESCRIPTION
**Pull request by mobuchowski**
This PR adds support for OpenLineage for FTPFileTransferOperator.

### CODE REVIEW

**Comment by pankajkoti** [comment]
cc: @utkarsharma2 could you please review this?

File: airflow/providers/ftp/operators/ftp.py [hunk 1]
```diff
@@ -135,6 +137,63 @@ def execute(self, context: Any) -> str | list[str] | None:
...
+
+        if self.operation.lower() == FTPOperation.GET:
+            inputs = remote_datasets
+            outputs = local_datasets
+        else:
+            inputs = local_datasets
+            outputs = remote_datasets
+
+        return OperatorLineage(
+            inputs=inputs,
```
Comment (sunank200) [inline airflow/providers/ftp/operators/ftp.py:191]: What about run and job facets?

**Review by sunank200 — COMMENTED** [review]
Documentation for the operator should be added for open lineage.

File: airflow/providers/ftp/operators/ftp.py (hunk 1 above)
Comment (mobuchowski) [reply airflow/providers/ftp/operators/ftp.py:191]: I don't see any relevant ones we could attach. That does not mean we can't add them later.

Comment (mobuchowski) [reply airflow/providers/ftp/operators/ftp.py:191]: There's also not much in terms of actual dataset data other than name and namespace we could attach I think.

**Review by eladkal — APPROVED** [review]

**Pull request merged**